Source/Base/Base/Math/Spline.cpp

```diff
@@ -0,0 +1,261 @@
+                currentPortion.reserve(_step + 1);
+
+                for (u32 i = 0; i < _step + 1; i++)
+                {
+                    f32 t = tStep * static_cast<f32>(i);
+                    currentPortion.emplace_back(InterpolateCombined(t, ptr, n, _alpha));
+                }
+
+                _storage.AddPortion(currentPortion);
+            }
+        }
+
+        return true;
+    }
+
+    vec3* Spline::GetPtrPoints()
+    {
+        if (_points.empty())
+            return nullptr;
+
+        vec3* ptr = new vec3[_points.size() * 3];
+        for (i32 i = 0; i < _points.size(); i++)
+        {
+            std::memcpy(ptr + (3 * i), _points[i].combined.data, 3 * sizeof(vec3));
+        }
+
+        return ptr;
+    }
+
+    bool Spline::IsInterpolatedWithControl()
+    {
+        switch (_type)
+        {
+        case InterpolationType::BSpline:
+        case InterpolationType::CatmullRom:
+        case InterpolationType::CatmullRom_Uniform:
+        case InterpolationType::CatmullRom_Centripetal:
+        case InterpolationType::CatmullRom_Chordal:
+            return false;
+        case InterpolationType::Hermite:
+        case InterpolationType::Bezier:
+        case InterpolationType::Linear:
+        default:
+            return true;
+        }
+    }
+
+    vec3 Spline::InterpolateControl(f32 t, const SplinePoint& start, const SplinePoint& end)
+    {
+        t = Math::Clamp(t, 0.f, 1.f);
+
+        switch (_type)
+        {
+        case InterpolationType::Linear:
+            return Interpolation::Linear::Lerp(t, start.control.point, end.control.point);
+        case InterpolationType::Bezier:
+            return Interpolation::Bezier::Cubic(t, start.control.point, start.control.exit, end.control.enter, end.control.point);
+        case InterpolationType::Hermite:
+            return Interpolation::Hermite::Cubic(t, start.control.point, end.control.point, start.control.exit, end.control.enter);
+        default:
+            return start.control.point;
+        }
+    }
+
+    vec3 Spline::InterpolateCombined(f32 t, const vec3* points, const i32 index, f32 alpha)
+    {
+        t = Math::Clamp(t, 0.f, 1.f);
+        alpha = Math::Clamp(alpha, 0.f, 1.f);
+
+        switch (_type)
+        {
+        case InterpolationType::BSpline:
+            return Interpolation::BSpline::BasisSpline(t, points, index);
+        case InterpolationType::CatmullRom:
+            return Interpolation::CatmullRom::Base(t, alpha, points, index);
+        case InterpolationType::CatmullRom_Uniform:
+            return Interpolation::CatmullRom::Uniform(t, points, index);
+        case InterpolationType::CatmullRom_Centripetal:
+            return Interpolation::CatmullRom::Centripetal(t, points, index);
+        case InterpolationType::CatmullRom_Chordal:
+            return Interpolation::CatmullRom::Chordal(t, points, index);
+        default:
+            break;
+        }
+
+        return *(points + index);
```

Comment:
Please make the result into a variable above, so that it is more clear what is happening here, also you can simply do (points[index]) as far as I am aware.